src/Components/WebAssembly/WebAssembly.Authentication/src/Models/RemoteUserAccount.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// A user account.
+    /// </summary>
+    /// <remarks>
+    /// The information in this type will be use to produce a <see cref="System.Security.Claims.ClaimsPrincipal"/> for the application.
+    /// </remarks>
+    public class RemoteUserAccount
+    {
+        /// <summary>
+        /// Gets or sets properties not explicitly mapped about the user.
+        /// </summary>
+        [JsonExtensionData]
+        public IDictionary<string, object> AdditionalProperties { get; set; }
```

Comment:
Is there a reason we couldn't deserialize a `IDictionary<string, string>` instead? Usually `JsonExtensionData` is useful if you have other properties on the type that you want to deserialize.

Reply:
I think the expected usage pattern is that people subclass `RemoteUserAccount` to define their own properties that should get deserialized and for which they have particular logic.

Then, the remaining properties get hoovered up by the framework into `AdditionalProperties` (though I don't know what the framework does with them).

Reply:
Steve is right, the idea here is that we want to allow people to create their own account types to represent the information specific to their auth providers and have a strongly typed model, while at the same time we provide a basic experience that just shoves everything into a property bag.